src/base/bittorrent/torrentimpl.cpp

```diff
@@ -1570,6 +1588,10 @@ void TorrentImpl::endReceivedMetadataHandling(const Path &savePath, const PathLi
     if (m_hasFirstLastPiecePriority)
         applyFirstLastPiecePriority(true);
 
+    // Check file name blacklist when metadata was not available in Session::addTorrent_impl
+    if (!hasFilePriorities)
+        applyExcludedFileNamesFilters();
+
```

Comment:
This is wrong approach!
File priorities should be modified before `reload()` is called. So they will applied to reloaded torrent from the scratch.